packages/ember-testing/lib/test.js

```diff
@@ -464,6 +465,58 @@ EmberApplication.reopen({
       delete this.testHelpers[name];
       delete this.originalMethods[name];
     }
+  },
+
+  /**
+    Builds a new instance of the application and registers test helpers on it.
+
+    Example:
+
+    ```javascript
+    var instance = run(App, 'buildTestInstance');
+    ```
+
+    @public
+    @method buildTestInstance
+  */
+  buildTestInstance() {
+    var instance = this.buildInstance();
+    this.runInstanceInitializers(instance);
+    instance.injectTestHelpers();
+    return instance;
+  }
+});
+
+EmberApplicationInstance.reopen({
+  testHelpers: null,
+
+  injectTestHelpers() {
+    this.reopen({
+      willDestroy() {
+        this._super(...arguments);
+        this.removeTestHelpers();
+      }
+    });
+
+    this.testHelpers = {};
+    for (var name in helpers) {
+      this[name] = this.testHelpers[name] = helper(this, name);
+    }
+
+    for (var i = 0, l = injectHelpersCallbacks.length; i < l; i++) {
+      injectHelpersCallbacks[i](this);
+    }
+  },
+
+  removeTestHelpers() {
+    if (!this.testHelpers) { return; }
+
+    for (var name in helpers) {
+      delete this.testHelpers[name];
```

Comment:
If we are going to set `this.testHelpers` to `null` do we actually need to `delete this.testHelpers[name]`?